store/src/server/api.rs

```diff
@@ -190,11 +221,15 @@ impl api_server::Api for StoreApi {
             .accounts
             .into_iter()
             .map(|account_update| {
+                let account_details: Option<AccountDetails> =
+                    <Option<AccountDetails>>::read_from_bytes(&account_update.details)
+                        .map_err(|err| Status::invalid_argument(err.to_string()))?;
```

Comment:
This should have validation:

- if the account is on-chain, there must be details
- if the account is off-chain, there must NOT be any details